otelcollector/configuration-reader-builder/main.go

```diff
@@ -48,6 +48,7 @@ type OtelConfig struct {
 		Prometheus struct {
 			Config          map[string]interface{} `yaml:"config"`
 			TargetAllocator interface{}            `yaml:"target_allocator"`
+			ApiServer       interface{}            `yaml:"api_server"`
```

Comment:
Hi Grace, This needs to be added in a few other places - prom-config-validator, tomlparser-set-global-config - etc. pls check

Reply:
Thanks Rashmi, added to these places. I'm also testing out having this in the shared golang package and everything referencing it so that we have just one place for it